### PR TITLE
[INV-43] Accommodation filter system with multiple criteria

### PR DESCRIPTION
**Pull request by Savesh1508**
https://github.com/user-attachments/assets/27a8f103-ffd7-401f-ad43-9e84bd9d8e1a

Available filters:
"min-price": 0,
  "max-price": 0,
  "search": "luxury",
  "street": "string",
  "city": "New York",
  "country": "string",
  "available": true,
  "available-from": "2024-11-15",
  "available-to": "2024-12-15",
  "rooms": 2,
  "has-wifi": true,
  "has-parking": true,
  "has-swimming-pool": true,
  "has-pet-allowance": true,
  "has-backyard": true,
  "has-smoking-allowance": true,
  "has-hospital-nearby": true,
  "has-laundary-service": true,
  "has-kitchen": true,
  "has-air-conditioning": true,
  "has-tv": true,
  "has-airport-transfer": true,
  "is-close-to-center": true,
  "is-child-friendly": true,
  "is-quiet-area": true

### CODE REVIEW

File: src/modules/accommodations/dto/accomodation-filters.dto.ts
Comment (SurajSulten) [inline src/modules/accommodations/dto/accomodation-filters.dto.ts:1]: write double 'm' in the name of the file 'accommodation-filters.dto.ts'

File: src/modules/accommodations/helpers/accomodation-filters.util.ts
Comment (SurajSulten) [inline src/modules/accommodations/helpers/accomodation-filters.util.ts:1]: double 'm' in the name of the file

**Review by SurajSulten — CHANGES_REQUESTED** [review]

File: src/modules/accommodations/services/accommodations.service.ts [hunk 1]
```diff
@@ -1,19 +1,42 @@
...
+    private readonly accommodationRepo: Repository<Accommodation>,
   ) {}
 
   create() {
     return 'This action adds a new Accommodation';
   }
 
-  findAll() {
-    return `This action returns all Accommodation`;
+  async getAllAccommodations(filters: AccommodationFiltersQueryDto) {
```
Comment (NegatiV-64) [inline src/modules/accommodations/services/accommodations.service.ts:25]: Rename to list

File: src/shared/interceptors/snake-to-camel.interceptor.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+@Injectable()
+export class SnakeToCamelInterceptor implements NestInterceptor {
+  intercept(ctx: ExecutionContext, next: CallHandler<any>): Observable<any> {
+    const request = ctx.switchToHttp().getRequest();
+
+    if (request.query) {
+      request.query = mapKeys(request.query, (value, key) => {
+        if (/[-_]{2,}|[^a-z0-9_-]/.test(key)) {
+          //Ignoring query if it's not in required format
```
Comment (NegatiV-64) [inline src/shared/interceptors/snake-to-camel.interceptor.ts:18]: Remove the comment

File: src/shared/interceptors/snake-to-camel.interceptor.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import { mapKeys, camelCase } from 'lodash';
+
+@Injectable()
+export class SnakeToCamelInterceptor implements NestInterceptor {
+  intercept(ctx: ExecutionContext, next: CallHandler<any>): Observable<any> {
+    const request = ctx.switchToHttp().getRequest();
+
+    if (request.query) {
+      request.query = mapKeys(request.query, (value, key) => {
+        if (/[-_]{2,}|[^a-z0-9_-]/.test(key)) {
```
Comment (NegatiV-64) [inline src/shared/interceptors/snake-to-camel.interceptor.ts:17]: What does it mean?

File: src/shared/utils/transform-type.util.ts
Comment (NegatiV-64) [inline src/shared/utils/transform-type.util.ts:1]: It is not a util, it is a transformer

**Review by NegatiV-64 — CHANGES_REQUESTED** [review]

File: src/modules/accommodations/dto/accomodation-filters.dto.ts
Comment (Savesh1508) [reply src/modules/accommodations/dto/accomodation-filters.dto.ts:1]: Done

File: src/modules/accommodations/entities/accommodations.entity.ts [hunk 4]
```diff
@@ -13,6 +13,7 @@ import { AccommodationImage } from './accommodation-image.entity';
 import { AccommodationAmenity } from './accommodation-amenity.entity';
 import { AccommodationAddress } from './accommodation-address.entity';
 import { Account } from '#/modules/user/entities/account.entity';
+import { AccommodationStatus } from '../../../shared/constants/accommodation-status.constants';
```
[the remark this answers is not in the document]
Comment (Savesh1508) [reply src/modules/accommodations/entities/accommodations.entity.ts:16]: Done

File: src/modules/accommodations/helpers/accomodation-filters.util.ts
Comment (Savesh1508) [reply src/modules/accommodations/helpers/accomodation-filters.util.ts:1]: Done

File: src/modules/accommodations/controllers/accommodations.controller.ts [hunk 5]
```diff
@@ -1,32 +1,58 @@
...
+    name: 'filters',
+    type: AccommodationFiltersQueryDto,
+    required: false,
+    description: 'Query parameters for filtering accommodations',
+  })
+  @ApiResponse({
+    status: 200,
+    description: 'List of accommodations fetched successfully',
+  })
+  async getAllAccommodations(@Query() filters: AccommodationFiltersQueryDto) {
```
[the remark this answers is not in the document]
Comment (Savesh1508) [reply src/modules/accommodations/controllers/accommodations.controller.ts:37]: Done

File: src/modules/accommodations/dto/accomodation-filters.dto.ts [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+  Min,
+} from 'class-validator';
+import {
+  TransformToBoolean,
+  TransformToDate,
+  TransformToNumber,
+} from '#/shared/utils/transform-type.util';
+
+export class AccommodationFiltersQueryDto {
+  // PRICE
```
[the remark this answers is not in the document]
Comment (Savesh1508) [reply src/modules/accommodations/dto/accomodation-filters.dto.ts:18]: Removed

File: src/modules/accommodations/dto/accomodation-filters.dto.ts [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+    description: 'Maximum price for accommodation',
+    required: false,
+    type: Number,
+    example: 500,
+  })
+  @IsOptional()
+  @IsNumber({}, { message: 'Maximum price must be a number' })
+  @Min(1, { message: 'Price maximum must be greater than or equal to 1' })
+  @TransformToNumber()
+  maxPrice?: number;
```
[the remark this answers is not in the document]
Comment (Savesh1508) [reply src/modules/accommodations/dto/accomodation-filters.dto.ts:43]: Get it, done

File: src/modules/accommodations/services/accommodations.service.ts (hunk 1 above)
Comment (Savesh1508) [reply src/modules/accommodations/services/accommodations.service.ts:25]: Done

File: src/shared/utils/transform-type.util.ts
Comment (Savesh1508) [reply src/shared/utils/transform-type.util.ts:1]: Replaced to transformers folder

File: src/shared/interceptors/snake-to-camel.interceptor.ts (hunk 3 above)
Comment (Savesh1508) [reply src/shared/interceptors/snake-to-camel.interceptor.ts:17]: It checks has query multiple separators or invalid characters.

For now, I make it more undestandable

File: src/shared/interceptors/snake-to-camel.interceptor.ts (hunk 2 above)
Comment (Savesh1508) [reply src/shared/interceptors/snake-to-camel.interceptor.ts:18]: Removed

**Review by NegatiV-64 — APPROVED** [review]

**Review by SurajSulten — APPROVED** [review]

**Pull request merged**